libs/framework/src/bundle_archive.c

```diff
@@ -28,644 +28,415 @@
-
-				struct dirent *dent = NULL;
-				struct stat st;
-
-                errno = 0;
-                dent = readdir(archive->archiveRootDir);
-				while (errno == 0 && dent != NULL) {
-					char subdir[512];
-					snprintf(subdir, 512, "%s/%s", archiveRoot, dent->d_name);
-					int rv = stat(subdir, &st);
-					if (rv == 0 && S_ISDIR(st.st_mode) && (strncmp(dent->d_name, "version", 7) == 0)) {
-						sscanf(dent->d_name, "version%*d.%ld", &idx);
-						if (idx > highestId) {
-							highestId = idx;
-						}
-					}
-                    errno = 0;
-                    dent = readdir(archive->archiveRootDir);
-				}
-
-				status = CELIX_DO_IF(status, bundleArchive_getRevisionLocation(archive, 0, &location));
-				status = CELIX_DO_IF(status, bundleArchive_reviseInternal(archive, true, highestId, location, NULL));
-				if (location) {
-					free(location);
-				}
-				if (status == CELIX_SUCCESS) {
-					*bundle_archive = archive;
-				}
-				closedir(archive->archiveRootDir);
-			}
-		}
-	}
-
-	if(status != CELIX_SUCCESS && archive != NULL){
-		bundleArchive_destroy(archive);
-	}
-
-	framework_logIfError(fw->logger, status, NULL, "Could not create archive");
-
-	return status;
-}
-
 celix_status_t bundleArchive_getId(bundle_archive_pt archive, long *id) {
-	celix_status_t status = CELIX_SUCCESS;
-
-	if (archive->id < 0) {
-		FILE *bundleIdFile;
-		char id[256];
-		char bundleId[512];
-		snprintf(bundleId, sizeof(bundleId), "%s/bundle.id", archive->archiveRoot);
-
-		bundleIdFile = fopen(bundleId, "r");
-		if(bundleIdFile!=NULL){
-			fgets(id, sizeof(id), bundleIdFile);
-			fclose(bundleIdFile);
-			sscanf(id, "%ld", &archive->id);
-		}
-		else{
-			status = CELIX_FILE_IO_EXCEPTION;
-		}
-	}
-
-	if (status == CELIX_SUCCESS) {
-		*id = archive->id;
-	}
+     *id = archive->id;
+	return CELIX_SUCCESS;
+}
 
-	framework_logIfError(archive->fw->logger, status, NULL, "Could not get archive id");
+long celix_bundleArchive_getId(bundle_archive_pt archive) {
+    return archive->id;
+}
 
-	return status;
+const char* celix_bundleArchive_getSymbolicName(bundle_archive_pt archive) {
+    return archive->bundleSymbolicName;
 }
 
 celix_status_t bundleArchive_getLocation(bundle_archive_pt archive, const char **location) {
-	celix_status_t status = CELIX_SUCCESS;
-	if (archive->location == NULL) {
-		FILE *bundleLocationFile;
-		char bundleLocation[512];
-		char loc[256];
-
-		snprintf(bundleLocation, sizeof(bundleLocation), "%s/bundle.location", archive->archiveRoot);
-
-		bundleLocationFile = fopen(bundleLocation, "r");
-		if(bundleLocationFile!=NULL){
-			fgets(loc, sizeof(loc), bundleLocationFile);
-			fclose(bundleLocationFile);
-			archive->location = strdup(loc);
-		}
-		else{
-			status = CELIX_FILE_IO_EXCEPTION;
-		}
-	}
-
-	if (status == CELIX_SUCCESS) {
-		*location = archive->location;
-	}
-
-	framework_logIfError(archive->fw->logger, status, NULL, "Could not get archive location");
-
-	return status;
+    *location = archive->location;
```

Comment:
`bundleArchive_revise` changes `archive->location`, so there is subtle thread-safety issue.